src/common/src/telemetry/report.rs

```diff
@@ -91,9 +101,29 @@ where
                 )
             });
 
+        let (tx, mut event_rx) = tokio::sync::mpsc::unbounded_channel::<PbEventMessage>();
+        TELEMETRY_EVENT_REPORT_TX.set(tx).unwrap_or_else(|_| {
+            tracing::warn!(
+                "Telemetry failed to set event reporting tx, event reporting will be disabled"
+            );
+        });
+
         loop {
             tokio::select! {
                 _ = interval.tick() => {},
+                event = event_rx.recv() => {
+                    if let Some(event) = event {
+                        TELEMETRY_EVENT_REPORT_STASH.lock().await.push(event);
+                    }
+                    if TELEMETRY_EVENT_REPORT_STASH.lock().await.len() >= TELEMETRY_EVENT_REPORT_STASH_SIZE {
```

Comment:
`TELEMETRY_EVENT_REPORT_STASH` with locking is unnecessary because `do_telemetry_event_report` is called in this tokio task only and there is no real contention on it. We can define a local var for the message batch and pass it to `do_telemetry_event_report` directly.
```rust
let mut batch = Vec::new();

loop {
  ....
  tokio::select! {
    ...
    ... => {
     ...
     do_telemetry_event_report(std::mem::take(&mut batch)).await;
    }
    ...
  }
}

Reply:
My intention is to move all event report stuff inside the `telemetry_event` crate, and defining a local var here may make the two telemetry object mixed. 
But yes, a local var does better here.